File: modules/global/src/com/haulmont/addon/datagen/entity/enm/EnumPropGenSettings.java
package com.haulmont.addon.datagen.entity.enm;

import com.haulmont.addon.datagen.entity.PropertyGenerationSettings;
import com.haulmont.chile.core.annotations.MetaClass;
import com.haulmont.chile.core.annotations.MetaProperty;

@MetaClass(name = "datagen_EnumPropGenSettings")
public class EnumPropGenSettings extends PropertyGenerationSettings {
    private static final long serialVersionUID = 4050643602429340299L;

    @MetaProperty
    protected String strategy = EnumPropGenStrategy.RANDOM.getId();

    protected Enum manualValue;

    public EnumPropGenStrategy getStrategy() {
        return strategy == null ? null : EnumPropGenStrategy.fromId(strategy);
    }

    public void setStrategy(EnumPropGenStrategy strategy) {
        this.strategy = strategy == null ? null : strategy.getId();
    }

    public Enum getManualValue() {
        return manualValue;
    }

    public void setManualValue(Enum manualValue) {
        this.manualValue = manualValue;
    }
}

File: modules/global/src/com/haulmont/addon/datagen/entity/enm/EnumPropGenSettings.kt
package com.haulmont.addon.datagen.entity.enm

import com.haulmont.addon.datagen.entity.PropertyGenerationSettings
import com.haulmont.chile.core.annotations.MetaClass
import com.haulmont.chile.core.annotations.MetaProperty

@MetaClass(name = "datagen_EnumPropGenSettings")
class EnumPropGenSettings : PropertyGenerationSettings() {

    @MetaProperty
    private var strategy: String? = null

    var manualValue: Enum<*>? = null

    fun getStrategy(): EnumPropGenStrategy? = strategy?.let { EnumPropGenStrategy.fromId(it) }

    fun setStrategy(strategy: EnumPropGenStrategy?) {
        this.strategy = strategy?.id
    }

    companion object {
        private const val serialVersionUID = -7094018282292550489L
    }
}
